Adds PID to file name, writes to a temp file.

gauravjsingh opened the pull request:
Adds the pid to the file name to reduce the likelihood that a file is
overwritten when suricata is unexpectedly killed.  The number in the
waldo file is only written out during a clean shutdown.  In the event
of an improper shutdown, extracted files will be written using the old
number and existing files with the same name will be overwritten.

Writes extracted files and their metadata to a temporary file prefixed
with a '.'.  Renames the files when they are completely done being
written and the meta file is closed.  As-is there is no way to know that
a file on disk is still being written to by suricata.

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
-
-
-

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

inliniac commented on src/log-filestore.c:
```diff
@@ -22,6 +22,9 @@
  *
  */
 
+#include <sys/types.h>
```
these should be unnecessary as they are included in suricata-common.h, so please remove them here

inliniac commented on src/log-filestore.c:
```diff
@@ -336,8 +348,19 @@ static void LogFilestoreLogCloseMetaFile(const File *ff)
+        /* Move working files to their final location now that we are done
+         * writing them.*/
+        if (rename(working_filename, final_filename) != 0) {
+            SCLogInfo("renaming file %s to %s failed", working_filename,
```
these should probably be warnings? OTOH we generally don't want to create output at pkt runtime for the risk of flooding. That risk should be quite low here though.

inliniac commented:
Any thoughts on making this optional?

I wonder if there might be use cases that depend on the current behavior. Can't really think of any, most ppl probably wait with their processing until the file is complete.

gauravjsingh commented:
Sorry for the delay. I'll open a new PR after addressing the comments.

For the last comment, should the option be to add PID and prefix, or do you think they should be separate options?
